app/models/annotation.rb

```diff
@@ -0,0 +1,15 @@
+class Annotation < ApplicationRecord
+  has_one :map_overlay
```

Comment:
These should be `belongs_to` not `has_one` - the annotation should have foreign key columns called `map_overlay_id` and `building_id`. Also the validations should ensure only one annotation per building/map layer combination.

Reply:
Also the foreign key should be `null: false` and validate the presence of both map layer and annotation.